zrml/neo-swaps/src/lib.rs

```diff
@@ -834,7 +834,7 @@ mod pallet {
             ensure!(market.scoring_rule == ScoringRule::Lmsr, Error::<T>::InvalidTradingMechanism);
             let asset_count = spot_prices.len();
             ensure!(asset_count as u16 == market.outcomes(), Error::<T>::IncorrectVecLen);
-            ensure!(market.outcomes() <= MAX_ASSETS, Error::<T>::AssetCountAboveMax);
+            ensure!(market.outcomes() as u32 <= MaxAssets::get(), Error::<T>::AssetCountAboveMax);
```

Comment:
Casting might result in unexpected results, for example `300u16 as u8` results in `44`.

```suggestion
            ensure!(market.outcomes().saturated_into() <= MaxAssets::get(), Error::<T>::AssetCountAboveMax);
```

Reply:
I get that, but we're casting to `u32` here, so that should always be fine. That being said, is there any argument against just using saturated or checked conversions instead? Seems like these casts are a risk in case we ever change the type of `market.outcomes()`. Also, clippy doesn't appear to warn on narrow casts.

Reply:
The difference is that the cast's proper functionality relies on an assumption, whereas the saturated conversion does not, as it just maxes out when an overflow occurs.

Reply:
Really doesn't matter in this case, I agree. But I tend to opt for the side of less conditions to remember.

Reply:
Went a little crazy and cleaned up all the conversions. The `NarrowingConversion` error is to catch vectors that are too large and would cause saturation. I understand this cannot happen for various other technical reasons, but somehow I felt this was necessary in terms of correctness.

Reply:
5d75af2b1f81d6cf0581d45b6fe00cc6df1f9ea1